<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Chrysostomos Tornari">
<meta name="dcterms.date" content="2023-02-23">

<title>TCGA_HN_Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="TCGA_HN_Analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="TCGA_HN_Analysis_files/libs/quarto-html/quarto.js"></script>
<script src="TCGA_HN_Analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="TCGA_HN_Analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="TCGA_HN_Analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="TCGA_HN_Analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="TCGA_HN_Analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="TCGA_HN_Analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="TCGA_HN_Analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="TCGA_HN_Analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background">Background</a>
  <ul class="collapse">
  <li><a href="#disclaimer" id="toc-disclaimer" class="nav-link" data-scroll-target="#disclaimer">Disclaimer</a></li>
  <li><a href="#aim" id="toc-aim" class="nav-link" data-scroll-target="#aim">Aim</a></li>
  </ul></li>
  <li><a href="#tcga-head-neck-carcinoma-information" id="toc-tcga-head-neck-carcinoma-information" class="nav-link" data-scroll-target="#tcga-head-neck-carcinoma-information">TCGA Head &amp; Neck Carcinoma Information</a></li>
  <li><a href="#rnaseq" id="toc-rnaseq" class="nav-link" data-scroll-target="#rnaseq">RNAseq</a>
  <ul class="collapse">
  <li><a href="#hnsc-rnaseq-data" id="toc-hnsc-rnaseq-data" class="nav-link" data-scroll-target="#hnsc-rnaseq-data">HNSC RNAseq data</a></li>
  </ul></li>
  <li><a href="#rnaseq-normalization" id="toc-rnaseq-normalization" class="nav-link" data-scroll-target="#rnaseq-normalization">RNASeq Normalization</a>
  <ul class="collapse">
  <li><a href="#step-by-step" id="toc-step-by-step" class="nav-link" data-scroll-target="#step-by-step">Step-by-step</a></li>
  </ul></li>
  <li><a href="#pipeline-approach-to-analysis" id="toc-pipeline-approach-to-analysis" class="nav-link" data-scroll-target="#pipeline-approach-to-analysis">Pipeline approach to analysis</a></li>
  <li><a href="#visualization" id="toc-visualization" class="nav-link" data-scroll-target="#visualization">Visualization</a>
  <ul class="collapse">
  <li><a href="#pca-plot" id="toc-pca-plot" class="nav-link" data-scroll-target="#pca-plot">PCA Plot</a></li>
  <li><a href="#heatmap" id="toc-heatmap" class="nav-link" data-scroll-target="#heatmap">Heatmap</a></li>
  <li><a href="#interpreting-heatmap" id="toc-interpreting-heatmap" class="nav-link" data-scroll-target="#interpreting-heatmap">Interpreting HeatMap</a></li>
  </ul></li>
  <li><a href="#further-heatmap-exercise" id="toc-further-heatmap-exercise" class="nav-link" data-scroll-target="#further-heatmap-exercise">Further heatmap exercise</a></li>
  <li><a href="#survival" id="toc-survival" class="nav-link" data-scroll-target="#survival">Survival</a>
  <ul class="collapse">
  <li><a href="#survival-by-gender" id="toc-survival-by-gender" class="nav-link" data-scroll-target="#survival-by-gender">Survival by gender</a></li>
  <li><a href="#survival-by-stage" id="toc-survival-by-stage" class="nav-link" data-scroll-target="#survival-by-stage">Survival by stage</a></li>
  </ul></li>
  <li><a href="#the-relationship-between-gene-expression-and-survival" id="toc-the-relationship-between-gene-expression-and-survival" class="nav-link" data-scroll-target="#the-relationship-between-gene-expression-and-survival">The relationship between gene expression and survival</a>
  <ul class="collapse">
  <li><a href="#getting-expression-results-from-the-limma-result" id="toc-getting-expression-results-from-the-limma-result" class="nav-link" data-scroll-target="#getting-expression-results-from-the-limma-result">Getting expression results from the limma result</a></li>
  <li><a href="#influence-of-expression-on-prognosis" id="toc-influence-of-expression-on-prognosis" class="nav-link" data-scroll-target="#influence-of-expression-on-prognosis">Influence of expression on prognosis</a></li>
  </ul></li>
  <li><a href="#functionalising-the-effect-of-expression-on-survival-for-multiple-genes" id="toc-functionalising-the-effect-of-expression-on-survival-for-multiple-genes" class="nav-link" data-scroll-target="#functionalising-the-effect-of-expression-on-survival-for-multiple-genes">Functionalising the effect of expression on survival for multiple genes</a></li>
  <li><a href="#understanding-results" id="toc-understanding-results" class="nav-link" data-scroll-target="#understanding-results">Understanding results</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">TCGA_HN_Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Chrysostomos Tornari </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 23, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="background" class="level2">
<h2 class="anchored" data-anchor-id="background">Background</h2>
<section id="disclaimer" class="level3">
<h3 class="anchored" data-anchor-id="disclaimer">Disclaimer</h3>
<p>This analysis was inspired by <a href="https://www.youtube.com/watch?v=oyAn5B-vLus">this tutorial</a>. It led me to <a href="https://www.costalab.org/wp-content/uploads/2022/11/handout_day4.html#2_TCGA_data">this guide</a> which I have followed but adapted for H&amp;N cancers.</p>
</section>
<section id="aim" class="level3">
<h3 class="anchored" data-anchor-id="aim">Aim</h3>
<p>I want to try and gain insights from the cancer genome atlas which may be applicable to Head &amp; Neck cancer. A PubMed search on “(cancer genome atlas OR TCGA) AND neck AND cancer AND (SCC OR squamous)” on 06/02/2023 revealed 1407 papers so this kind of analysis is unlikely to represent anything too new.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Loading libraries_b4f4205d4cf76847bb739a265b09be19">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!requireNamespace('BiocManager', quietly = TRUE))</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     install.packages('BiocManager')</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('TCGAbiolinks')</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"TCGAbiolinks"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('limma')</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"limma"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('edgeR')</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"glmnet"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"factoextra"</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"FactoMineR"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"caret"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"SummarizedExperiment"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"gplots"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"survival"</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"survminer"</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"RColorBrewer"</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('clusterProfiler')</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"clusterProfiler"</span>)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install('genefilter')</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"genefilter"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="tcga-head-neck-carcinoma-information" class="level2">
<h2 class="anchored" data-anchor-id="tcga-head-neck-carcinoma-information">TCGA Head &amp; Neck Carcinoma Information</h2>
<p>In this analysis I will look at a number of Head &amp; Neck cancer data from the above database. Head &amp; Neck cancer is referred to as HNSC in TCGA. You can then filter by subsite and explore the data though different subsites have quite different sample sizes. This can be seen in <a href="https://portal.gdc.cancer.gov/projects/TCGA-HNSC">this table</a>.</p>
<p>The TCGA biolinks package installed and loaded in the above code block provides matching donors across different data tables and facilitates R analysis. There are some particularly important functions:</p>
<ul>
<li><code>GDCquery</code> allows you to investigate TCGA</li>
<li><code>GDCdownload</code> downloads raw versions of the desired files</li>
<li><code>GDCprepare</code> puts them into an R-analysable format</li>
</ul>
<p>You can use <code>head(getGDCprojects())</code> to look at the available projects but I already knwow the identity of the project I need from manually browsing <a href="https://portal.gdc.cancer.gov/">TCGA</a> and searching for H&amp;N. Here is a summary of the H&amp;N SCC data:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Pulling HN summary from TCGA_84d09a29e70c02acdf5da9bfccf3b19c">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>TCGAbiolinks<span class="sc">::</span><span class="fu">getProjectSummary</span>(<span class="st">"TCGA-HNSC"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$file_count
[1] 25804

$data_categories
  file_count case_count               data_category
1       4340        526       Copy Number Variation
2       3709        528            Sequencing Reads
3       7166        510 Simple Nucleotide Variation
4       1740        528             DNA Methylation
5       1103        528                    Clinical
6       2270        528     Transcriptome Profiling
7       2858        528                 Biospecimen
8        354        354          Proteome Profiling
9       2264        521        Structural Variation

$case_count
[1] 528

$file_size
[1] 7.172321e+13</code></pre>
</div>
</div>
</section>
<section id="rnaseq" class="level2">
<h2 class="anchored" data-anchor-id="rnaseq">RNAseq</h2>
<p>The example starts by looking at RNAseq. This is something relatively new to me as it’s a next-generation sequencing approach that gained popularity since my PhD. It is a technique that analyses the quantity and sequencing of RNA in a sample. It analyses the transcriptome. Although it sounds like you directly analyse RNA rather than using reverse transcriptase to generate cDNA, this is not true. cDNA is generated and next-generation sequencing applied. It is a quantitative technique as it can provide transcription levels.Finally, it also captures splice variants and post-transcriptional modifications. Filtering steps to deplete tRNA and rRNA can be applied to leave just mRNA if this is desired.</p>
<section id="hnsc-rnaseq-data" class="level3">
<h3 class="anchored" data-anchor-id="hnsc-rnaseq-data">HNSC RNAseq data</h3>
<p>As per the example, I will first look at the ‘Transcriptome Profiling’ within the population of interest. Though the example statest that we will use ‘HTSeq’ counts they actually use STAR Counts. STAR stands for Spliced Transcripts Alignment to a Reference. It is an accurate and fast method of aligning reads to a reference genome.For every read presented, this algorithm will search for the longest sequence that exactly matches the reference genome. It will then try to match the rest of the unmatched read (e.g.&nbsp;to the next exon in the genome). It includes strategies to manage mismatches and deletions.</p>
<p>I first got the STAR-counts for all HNSCCs in the database</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Querying TCGA_594cb03626b9df2996ddb96b26534cae">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># I changed an "=" in the below statement to "&lt;-"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>query_TCGA <span class="ot">&lt;-</span> <span class="fu">GDCquery</span>(</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">"TCGA-HNSC"</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.category =</span> <span class="st">"Transcriptome Profiling"</span>, <span class="co"># parameter enforced by GDCquery</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.type =</span> <span class="st">"Gene Expression Quantification"</span>,</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">experimental.strategy =</span> <span class="st">"RNA-Seq"</span>,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#workflow.type = "HTSeq - Counts"</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">workflow.type =</span> <span class="st">"STAR - Counts"</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>o GDCquery: Searching in GDC database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Genome of reference: hg38</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>oo Accessing GDC. This might take a while...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo Project: TCGA-HNSC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>oo Filtering results</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By experimental.strategy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By data.type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By workflow.type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>----------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>oo Checking data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>----------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo Checking if there are duplicated cases</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo Checking if there are results for the query</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>o Preparing output</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-------------------</code></pre>
</div>
</div>
<p>This returns a data frame which contains another dataframe in [1,1]. this is a summary of the columns in the [1,1] df:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Analysing column names_b7457fd3dd04a6bb4026e33504326aa5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">getResults</span>(query_TCGA))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "id"                        "data_format"              
 [3] "cases"                     "access"                   
 [5] "file_name"                 "submitter_id"             
 [7] "data_category"             "type"                     
 [9] "file_size"                 "created_datetime"         
[11] "md5sum"                    "updated_datetime"         
[13] "file_id"                   "data_type"                
[15] "state"                     "experimental_strategy"    
[17] "version"                   "data_release"             
[19] "project"                   "analysis_id"              
[21] "analysis_state"            "analysis_submitter_id"    
[23] "analysis_workflow_link"    "analysis_workflow_type"   
[25] "analysis_workflow_version" "sample_type"              
[27] "is_ffpe"                   "cases.submitter_id"       
[29] "sample.submitter_id"      </code></pre>
</div>
</div>
<p>On exploring the full dataframe I agree with the exercise that the type of samples are of interest:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Sample types_a89013929097af8cc7ec6fe832245821">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getResults</span>(query_TCGA) <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample_type) <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  hablar<span class="sc">::</span><span class="fu">convert</span>(hablar<span class="sc">::</span><span class="fu">fct</span>(sample_type)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="co"># Can also use table() as last statement</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>              sample_type 
 Metastatic         :  2  
 Primary Tumor      :520  
 Solid Tissue Normal: 44  </code></pre>
</div>
</div>
<p>Given the poor representation of metastases in this dataset there is no point in further analysing this set at the time of writing. The walk-through tells us how to re-generate the dataset without these. This seems very inefficient (as per a lot of the code) but I will do it this way to prevent future errors as I am unfamiliar with GDCquery…</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Get STAR counts of data_68aeef70b4f36db753256eb8064acd8f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>query_TCGA <span class="ot">&lt;-</span> <span class="fu">GDCquery</span>(</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">"TCGA-HNSC"</span>,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.category =</span> <span class="st">"Transcriptome Profiling"</span>, <span class="co"># parameter enforced by GDCquery</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.type =</span> <span class="st">"Gene Expression Quantification"</span>,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">experimental.strategy =</span> <span class="st">"RNA-Seq"</span>,</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#workflow.type = "HTSeq - Counts"</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">workflow.type =</span> <span class="st">"STAR - Counts"</span>,</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample.type =</span> <span class="fu">c</span>(<span class="st">"Primary Tumor"</span>, <span class="st">"Solid Tissue Normal"</span>) <span class="co"># beware American spelling</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>o GDCquery: Searching in GDC database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Genome of reference: hg38</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>oo Accessing GDC. This might take a while...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------------------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo Project: TCGA-HNSC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>oo Filtering results</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>--------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By experimental.strategy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By data.type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By workflow.type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo By sample.type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>----------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>oo Checking data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>----------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo Checking if there are duplicated cases</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ooo Checking if there are results for the query</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-------------------</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>o Preparing output</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-------------------</code></pre>
</div>
</div>
<p>I’ll just check that this new dataset has the right sample types:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Check sample types_04fe6a7318960e716825cd2b656cc2f3">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getResults</span>(query_TCGA) <span class="sc">%&gt;%</span> </span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample_type) <span class="sc">%&gt;%</span> </span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  hablar<span class="sc">::</span><span class="fu">convert</span>(hablar<span class="sc">::</span><span class="fu">fct</span>(sample_type)) <span class="sc">%&gt;%</span> </span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>              sample_type 
 Primary Tumor      :520  
 Solid Tissue Normal: 44  </code></pre>
</div>
</div>
<p>I am doing this analysis as a project so that I don’t have to worry about the working directory. When GDCdownload gets the relevant files to the above query (which is currently just a list of the files) it will put them in the working directory. If you have downloaded this project from GitHub then the files will be downloaded into the project directory where a new directory called “GDCdata” will be created. Currently the files for this particular analysis are not in the GitHub folder and you will have to download them yourself by running the above code (as they are too large to host).</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Downloading files_976410226202fc559484832618eb14e4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">GDCdownload</span>(<span class="at">query =</span> query_TCGA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Downloading data for project TCGA-HNSC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Of the 564 files for download 564 already exist.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>All samples have been already downloaded</code></pre>
</div>
</div>
<p>I will now read the data into r.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Prepare data_ef3bed056a3aafc0465e83aedd23259d">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that you can skip this step and see the next block if you have the </span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="co"># tcga_data.RDS file in your directory though this will not be available on </span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="co"># GitHub as too large. I'd have to remotely host this and provide the link for </span></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="co"># other users to be able to access this.</span></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>tcga_data <span class="ot">&lt;-</span> <span class="fu">GDCprepare</span>(query_TCGA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Starting to add information to samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code> =&gt; Add clinical information to samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code> =&gt; Adding TCGA molecular information from marker papers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code> =&gt; Information will have prefix 'paper_' </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>hnsc subtype information from:doi:10.1038/nature14129</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Available assays in SummarizedExperiment : 
  =&gt; unstranded
  =&gt; stranded_first
  =&gt; stranded_second
  =&gt; tpm_unstrand
  =&gt; fpkm_unstrand
  =&gt; fpkm_uq_unstrand</code></pre>
</div>
</div>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Save data to RDS_2d1fa40d56ccdb47b58771cda6e49ac3">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the data as a file, if you need it later, you can just load this file</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="co"># instead of having to run the whole pipeline again</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> tcga_data,</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">"tcga_data.RDS"</span>,</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">compress =</span> <span class="cn">FALSE</span>)</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a><span class="co"># tcga_data &lt;- readRDS(file = "tcga_data.RDS")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>For the selected H&amp;N SCC primary and normal samples we have rows and 564 columns. Here are the columns available:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Check colnames of data_5c29b32b0a80b02a9f39e120005fa35f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">colData</span>(tcga_data))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "barcode"                     "patient"                    
 [3] "sample"                      "shortLetterCode"            
 [5] "definition"                  "sample_submitter_id"        
 [7] "sample_type_id"              "oct_embedded"               
 [9] "sample_id"                   "submitter_id"               
[11] "state"                       "is_ffpe"                    
[13] "sample_type"                 "tissue_type"                
[15] "tumor_descriptor"            "composition"                
[17] "days_to_collection"          "initial_weight"             
[19] "intermediate_dimension"      "pathology_report_uuid"      
[21] "shortest_dimension"          "longest_dimension"          
[23] "synchronous_malignancy"      "ajcc_pathologic_stage"      
[25] "days_to_diagnosis"           "treatments"                 
[27] "last_known_disease_status"   "tissue_or_organ_of_origin"  
[29] "days_to_last_follow_up"      "age_at_diagnosis"           
[31] "primary_diagnosis"           "ajcc_clinical_stage"        
[33] "prior_malignancy"            "year_of_diagnosis"          
[35] "prior_treatment"             "ajcc_staging_system_edition"
[37] "ajcc_pathologic_t"           "morphology"                 
[39] "ajcc_clinical_m"             "ajcc_clinical_n"            
[41] "ajcc_clinical_t"             "classification_of_tumor"    
[43] "diagnosis_id"                "icd_10_code"                
[45] "site_of_resection_or_biopsy" "tumor_grade"                
[47] "progression_or_recurrence"   "ajcc_pathologic_n"          
[49] "ajcc_pathologic_m"           "cigarettes_per_day"         
[51] "alcohol_history"             "exposure_id"                
[53] "years_smoked"                "pack_years_smoked"          
[55] "race"                        "gender"                     
[57] "ethnicity"                   "vital_status"               
[59] "age_at_index"                "days_to_birth"              
[61] "year_of_birth"               "demographic_id"             
[63] "days_to_death"               "year_of_death"              
[65] "bcr_patient_barcode"         "primary_site"               
[67] "project_id"                  "disease_type"               
[69] "name"                        "releasable"                 
[71] "released"                    "paper_patient"              
[73] "paper_RNA"                   "paper_Methylation"          
[75] "paper_RPPA"                  "paper_miRNA"                
[77] "paper_Copy.Number"           "paper_PARADIGM"             </code></pre>
</div>
</div>
<p>Some useful figures:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/unnamed-chunk-12_ca8b7531136937cc5558adba834e21c3">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># label: Useful tables</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_data<span class="sc">@</span>colData<span class="sc">$</span>vital_status) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> Var1 </th>
   <th style="text-align:right;"> Freq </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> Alive </td>
   <td style="text-align:right;"> 310 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Dead </td>
   <td style="text-align:right;"> 254 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_data<span class="sc">@</span>colData<span class="sc">$</span>ajcc_pathologic_stage) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> Var1 </th>
   <th style="text-align:right;"> Freq </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> Stage I </td>
   <td style="text-align:right;"> 29 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Stage II </td>
   <td style="text-align:right;"> 87 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Stage III </td>
   <td style="text-align:right;"> 89 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Stage IVA </td>
   <td style="text-align:right;"> 270 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Stage IVB </td>
   <td style="text-align:right;"> 12 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Stage IVC </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_data<span class="sc">@</span>colData<span class="sc">$</span>definition) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> Var1 </th>
   <th style="text-align:right;"> Freq </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> Primary solid Tumor </td>
   <td style="text-align:right;"> 520 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Solid Tissue Normal </td>
   <td style="text-align:right;"> 44 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_data<span class="sc">@</span>colData<span class="sc">$</span>tissue_or_organ_of_origin) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> Var1 </th>
   <th style="text-align:right;"> Freq </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> Anterior floor of mouth </td>
   <td style="text-align:right;"> 2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Base of tongue, NOS </td>
   <td style="text-align:right;"> 26 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Border of tongue </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Cheek mucosa </td>
   <td style="text-align:right;"> 19 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Floor of mouth, NOS </td>
   <td style="text-align:right;"> 56 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Gum, NOS </td>
   <td style="text-align:right;"> 8 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Hard palate </td>
   <td style="text-align:right;"> 4 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Hypopharynx, NOS </td>
   <td style="text-align:right;"> 9 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Larynx, NOS </td>
   <td style="text-align:right;"> 127 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Lip, NOS </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Lower gum </td>
   <td style="text-align:right;"> 2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Mandible </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Mouth, NOS </td>
   <td style="text-align:right;"> 23 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Oropharynx, NOS </td>
   <td style="text-align:right;"> 9 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Overlapping lesion of lip, oral cavity and pharynx </td>
   <td style="text-align:right;"> 83 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Palate, NOS </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Pharynx, NOS </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Posterior wall of oropharynx </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Retromolar area </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Supraglottis </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Tongue, NOS </td>
   <td style="text-align:right;"> 140 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Tonsil, NOS </td>
   <td style="text-align:right;"> 44 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Upper Gum </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Ventral surface of tongue, NOS </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_data<span class="sc">@</span>colData<span class="sc">$</span>gender) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> Var1 </th>
   <th style="text-align:right;"> Freq </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> female </td>
   <td style="text-align:right;"> 150 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> male </td>
   <td style="text-align:right;"> 414 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_data<span class="sc">@</span>colData<span class="sc">$</span>race) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;"> Var1 </th>
   <th style="text-align:right;"> Freq </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> american indian or alaska native </td>
   <td style="text-align:right;"> 2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> asian </td>
   <td style="text-align:right;"> 12 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> black or african american </td>
   <td style="text-align:right;"> 50 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> not reported </td>
   <td style="text-align:right;"> 16 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> white </td>
   <td style="text-align:right;"> 484 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">assay</span>(tcga_data)) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:right;"> x </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;"> 60660 </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 564 </td>
  </tr>
</tbody>
</table>

</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(tcga_data)) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> source </th>
   <th style="text-align:left;"> type </th>
   <th style="text-align:right;"> score </th>
   <th style="text-align:right;"> phase </th>
   <th style="text-align:left;"> gene_id </th>
   <th style="text-align:left;"> gene_type </th>
   <th style="text-align:left;"> gene_name </th>
   <th style="text-align:left;"> level </th>
   <th style="text-align:left;"> hgnc_id </th>
   <th style="text-align:left;"> havana_gene </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000000003.15 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000000003.15 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> TSPAN6 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:11858 </td>
   <td style="text-align:left;"> OTTHUMG00000022002.2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000005.6 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000000005.6 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> TNMD </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:17757 </td>
   <td style="text-align:left;"> OTTHUMG00000022001.2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000419.13 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000000419.13 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> DPM1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3005 </td>
   <td style="text-align:left;"> OTTHUMG00000032742.2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000457.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000000457.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> SCYL3 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:19285 </td>
   <td style="text-align:left;"> OTTHUMG00000035941.6 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000460.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000000460.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> C1orf112 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:25565 </td>
   <td style="text-align:left;"> OTTHUMG00000035821.9 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000000938.13 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000000938.13 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FGR </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3697 </td>
   <td style="text-align:left;"> OTTHUMG00000003516.3 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
</section>
<section id="rnaseq-normalization" class="level2">
<h2 class="anchored" data-anchor-id="rnaseq-normalization">RNASeq Normalization</h2>
<section id="step-by-step" class="level3">
<h3 class="anchored" data-anchor-id="step-by-step">Step-by-step</h3>
<p>Here is a step-by-step version of a pipeline</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Step-by-step pipeline_ae0d7994b37af89be795b155eb624d63">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> <span class="fu">colData</span>(tcga_data)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(clinical_data<span class="sc">$</span>definition)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">relevel</span>(group, <span class="at">ref=</span><span class="st">"Solid Tissue Normal"</span>)</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="co"># head(design)</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove low count genes (&lt;10 by default)</span></span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a><span class="co"># First create a Differential Gene Expression list</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> <span class="fu">assay</span>(tcga_data),</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> <span class="fu">colData</span>(tcga_data),</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes =</span> <span class="fu">as.data.frame</span>(<span class="fu">rowData</span>(tcga_data))</span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering</span></span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge, design) <span class="co"># Define the genes to keep</span></span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[keep,,keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># filter the object</span></span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(keep)</span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge, <span class="at">metho =</span> <span class="st">"TMM"</span>) <span class="co"># TMM = trimmed mean of M-values</span></span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a><span class="co"># We then need to conver the data to have similar variance as arrays with the VOOM</span></span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a><span class="co"># method. This is presumably because limma was designed to run on arrays rather</span></span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a><span class="co"># than NextGen sequencing libraries</span></span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, design, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Step-by-step%20pipeline-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<blockquote class="blockquote">
<p>“Finally, using lmFit lets fit a series of linear models, one to each of the probes. These data will then be fed to eBayes to produce a complex object which holds a number of statistics that we can use to rank the differentially expressed genes.”</p>
</blockquote>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Manually fitting data_4081c46ba4a0f59f0897297af75f83d5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We will now look at the top 10 differentially expressed genes:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Identifying top genes_f10d91b47170a5a8d963ae9f3fdc9514">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>topGenes <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit, <span class="at">coef =</span> <span class="dv">1</span>, <span class="at">sort.by =</span> <span class="st">"p"</span>)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>topGenes <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> source </th>
   <th style="text-align:left;"> type </th>
   <th style="text-align:right;"> score </th>
   <th style="text-align:right;"> phase </th>
   <th style="text-align:left;"> gene_id </th>
   <th style="text-align:left;"> gene_type </th>
   <th style="text-align:left;"> gene_name </th>
   <th style="text-align:left;"> level </th>
   <th style="text-align:left;"> hgnc_id </th>
   <th style="text-align:left;"> havana_gene </th>
   <th style="text-align:right;"> logFC </th>
   <th style="text-align:right;"> AveExpr </th>
   <th style="text-align:right;"> t </th>
   <th style="text-align:right;"> P.Value </th>
   <th style="text-align:right;"> adj.P.Val </th>
   <th style="text-align:right;"> B </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000004142.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000004142.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> POLDIP2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:23781 </td>
   <td style="text-align:left;"> OTTHUMG00000132065.9 </td>
   <td style="text-align:right;"> 7.149473 </td>
   <td style="text-align:right;"> 6.946990 </td>
   <td style="text-align:right;"> 117.50811 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 905.9611 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000004455.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000004455.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> AK2 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:362 </td>
   <td style="text-align:left;"> OTTHUMG00000004131.7 </td>
   <td style="text-align:right;"> 6.822341 </td>
   <td style="text-align:right;"> 6.992407 </td>
   <td style="text-align:right;"> 92.38328 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 777.4261 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000004487.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000004487.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> KDM1A </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:29079 </td>
   <td style="text-align:left;"> OTTHUMG00000003220.7 </td>
   <td style="text-align:right;"> 6.341754 </td>
   <td style="text-align:right;"> 6.825723 </td>
   <td style="text-align:right;"> 90.08590 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 764.0778 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000005007.13 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000005007.13 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> UPF1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:9962 </td>
   <td style="text-align:left;"> OTTHUMG00000183025.2 </td>
   <td style="text-align:right;"> 6.948789 </td>
   <td style="text-align:right;"> 7.115216 </td>
   <td style="text-align:right;"> 104.45844 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 842.7311 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000005022.6 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000005022.6 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> SLC25A5 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:10991 </td>
   <td style="text-align:left;"> OTTHUMG00000022715.3 </td>
   <td style="text-align:right;"> 8.376831 </td>
   <td style="text-align:right;"> 8.459095 </td>
   <td style="text-align:right;"> 93.97093 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 786.5924 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000005175.10 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000005175.10 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> RPAP3 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:26151 </td>
   <td style="text-align:left;"> OTTHUMG00000169668.3 </td>
   <td style="text-align:right;"> 4.714118 </td>
   <td style="text-align:right;"> 4.806713 </td>
   <td style="text-align:right;"> 85.67433 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 737.3424 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000005700.15 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000005700.15 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> IBTK </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:17853 </td>
   <td style="text-align:left;"> OTTHUMG00000015102.3 </td>
   <td style="text-align:right;"> 6.074055 </td>
   <td style="text-align:right;"> 5.448985 </td>
   <td style="text-align:right;"> 85.16925 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 734.5915 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000005893.16 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000005893.16 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> LAMP2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:6501 </td>
   <td style="text-align:left;"> OTTHUMG00000022301.5 </td>
   <td style="text-align:right;"> 7.932924 </td>
   <td style="text-align:right;"> 7.925460 </td>
   <td style="text-align:right;"> 95.65861 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 795.9893 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000006007.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000006007.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> GDE1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:29644 </td>
   <td style="text-align:left;"> OTTHUMG00000131454.7 </td>
   <td style="text-align:right;"> 6.217810 </td>
   <td style="text-align:right;"> 5.829674 </td>
   <td style="text-align:right;"> 96.79175 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 802.0044 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000006451.8 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000006451.8 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> RALA </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:9839 </td>
   <td style="text-align:left;"> OTTHUMG00000128775.5 </td>
   <td style="text-align:right;"> 6.713204 </td>
   <td style="text-align:right;"> 6.603064 </td>
   <td style="text-align:right;"> 86.87866 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 745.0809 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
</section>
<section id="pipeline-approach-to-analysis" class="level2">
<h2 class="anchored" data-anchor-id="pipeline-approach-to-analysis">Pipeline approach to analysis</h2>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Functional pipeline approach_8bbbfab53b7955d855a00ad68da1a986">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>limma_pipeline <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>  tcga_data,</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  condition_variable,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference_group=</span><span class="cn">NULL</span>){</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>  design_factor <span class="ot">=</span> <span class="fu">colData</span>(tcga_data)[, condition_variable, drop<span class="ot">=</span>T]</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>  group <span class="ot">=</span> <span class="fu">factor</span>(design_factor)</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(reference_group)){group <span class="ot">=</span> <span class="fu">relevel</span>(group, <span class="at">ref=</span>reference_group)}</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>  design <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> group)</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>  dge <span class="ot">=</span> <span class="fu">DGEList</span>(<span class="at">counts=</span><span class="fu">assay</span>(tcga_data),</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">samples=</span><span class="fu">colData</span>(tcga_data),</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>                 <span class="at">genes=</span><span class="fu">as.data.frame</span>(<span class="fu">rowData</span>(tcga_data)))</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filtering</span></span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">=</span> <span class="fu">filterByExpr</span>(dge,design)</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>  dge <span class="ot">=</span> dge[keep,,keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(keep)</span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Normalization (TMM followed by voom)</span></span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a>  dge <span class="ot">=</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a>  v <span class="ot">=</span> <span class="fu">voom</span>(dge, design, <span class="at">plot=</span><span class="cn">TRUE</span>)</span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Fit model to data given design</span></span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">=</span> <span class="fu">lmFit</span>(v, design)</span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">=</span> <span class="fu">eBayes</span>(fit)</span>
<span id="cb82-29"><a href="#cb82-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-30"><a href="#cb82-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Show top genes</span></span>
<span id="cb82-31"><a href="#cb82-31" aria-hidden="true" tabindex="-1"></a>  topGenes <span class="ot">=</span> <span class="fu">topTable</span>(fit, <span class="at">coef=</span><span class="fu">ncol</span>(design), <span class="at">number=</span><span class="dv">100</span>, <span class="at">sort.by=</span><span class="st">"p"</span>)</span>
<span id="cb82-32"><a href="#cb82-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-33"><a href="#cb82-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(</span>
<span id="cb82-34"><a href="#cb82-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(</span>
<span id="cb82-35"><a href="#cb82-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">voomObj=</span>v, <span class="co"># normalized data</span></span>
<span id="cb82-36"><a href="#cb82-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">fit=</span>fit, <span class="co"># fitted model and statistics</span></span>
<span id="cb82-37"><a href="#cb82-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">topGenes=</span>topGenes <span class="co"># the 100 most differentially expressed genes</span></span>
<span id="cb82-38"><a href="#cb82-38" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb82-39"><a href="#cb82-39" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb82-40"><a href="#cb82-40" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now with a pipeline, we can easily make different queries. Firstly by tissue type:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Tissue type through pipeline_22f0585794ed6d6cc2dd0d54b7c5a223">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>limma_res <span class="ot">&lt;-</span> <span class="fu">limma_pipeline</span>(</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">tcga_data=</span>tcga_data,</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_variable=</span><span class="st">"definition"</span>,</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference_group=</span><span class="st">"Solid Tissue Normal"</span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Tissue%20type%20through%20pipeline-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This result is saved also:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Save RDS of pipeline output_4c0da8957740fea5a1472c48f7dca03b">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the data as a file, if you need it later, you can just load this file</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co"># instead of having to run the whole pipeline again</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> limma_res,</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">"limma_res.RDS"</span>,</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">compress =</span> <span class="cn">FALSE</span>)</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a><span class="co"># limma_res &lt;- readRDS(file = "limma_res.RDS")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here is another analysis by gender</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Gender pipeline analysis_851c93ee23496c0dad52c1cec1159f63">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>gender_limma_res <span class="ot">&lt;-</span> <span class="fu">limma_pipeline</span>(</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">tcga_data=</span>tcga_data,</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_variable=</span><span class="st">"gender"</span>,</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference_group=</span><span class="st">"female"</span></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Gender%20pipeline%20analysis-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="at">object =</span> gender_limma_res,</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="st">"gender_limma_res.RDS"</span>,</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">compress =</span> <span class="cn">FALSE</span>)</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="co"># gender_limma_res &lt;- readRDS(file = "gender_limma_res.RDS")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s look at the top genes that come out of these analyses firstly by tissue type (normal vs cancer):</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Pipeline top genes by tissue_102e2639ccd0b1810b99025e56207733">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(limma_res<span class="sc">$</span>topGenes, <span class="dv">20</span>) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> source </th>
   <th style="text-align:left;"> type </th>
   <th style="text-align:right;"> score </th>
   <th style="text-align:right;"> phase </th>
   <th style="text-align:left;"> gene_id </th>
   <th style="text-align:left;"> gene_type </th>
   <th style="text-align:left;"> gene_name </th>
   <th style="text-align:left;"> level </th>
   <th style="text-align:left;"> hgnc_id </th>
   <th style="text-align:left;"> havana_gene </th>
   <th style="text-align:right;"> logFC </th>
   <th style="text-align:right;"> AveExpr </th>
   <th style="text-align:right;"> t </th>
   <th style="text-align:right;"> P.Value </th>
   <th style="text-align:right;"> adj.P.Val </th>
   <th style="text-align:right;"> B </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000181092.10 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000181092.10 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ADIPOQ </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:13633 </td>
   <td style="text-align:left;"> OTTHUMG00000156521.3 </td>
   <td style="text-align:right;"> -7.176794 </td>
   <td style="text-align:right;"> -4.7711913 </td>
   <td style="text-align:right;"> -27.13784 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 227.1603 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000096006.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000096006.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> CRISP3 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:16904 </td>
   <td style="text-align:left;"> OTTHUMG00000014823.3 </td>
   <td style="text-align:right;"> -8.591856 </td>
   <td style="text-align:right;"> -1.0019402 </td>
   <td style="text-align:right;"> -26.60984 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 221.6612 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000196616.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000196616.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ADH1B </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:250 </td>
   <td style="text-align:left;"> OTTHUMG00000161413.4 </td>
   <td style="text-align:right;"> -6.608842 </td>
   <td style="text-align:right;"> -2.3444701 </td>
   <td style="text-align:right;"> -24.26854 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 193.8269 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000034971.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000034971.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> MYOC </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:7610 </td>
   <td style="text-align:left;"> OTTHUMG00000034789.4 </td>
   <td style="text-align:right;"> -6.335138 </td>
   <td style="text-align:right;"> -4.4762472 </td>
   <td style="text-align:right;"> -23.98936 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 190.0965 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000102547.19 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000102547.19 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> CAB39L </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:20290 </td>
   <td style="text-align:left;"> OTTHUMG00000016914.6 </td>
   <td style="text-align:right;"> -2.238875 </td>
   <td style="text-align:right;"> 2.6534008 </td>
   <td style="text-align:right;"> -23.87185 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 189.2265 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000167588.13 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000167588.13 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> GPD1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:4455 </td>
   <td style="text-align:left;"> OTTHUMG00000169813.6 </td>
   <td style="text-align:right;"> -4.683988 </td>
   <td style="text-align:right;"> -0.4668095 </td>
   <td style="text-align:right;"> -23.86213 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 189.0425 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000166819.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000166819.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> PLIN1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:9076 </td>
   <td style="text-align:left;"> OTTHUMG00000149813.5 </td>
   <td style="text-align:right;"> -4.974732 </td>
   <td style="text-align:right;"> -2.4730633 </td>
   <td style="text-align:right;"> -22.73266 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 175.4408 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000168309.18 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000168309.18 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FAM107A </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:30827 </td>
   <td style="text-align:left;"> OTTHUMG00000159159.5 </td>
   <td style="text-align:right;"> -3.810493 </td>
   <td style="text-align:right;"> 1.1439306 </td>
   <td style="text-align:right;"> -22.47880 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 172.6963 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198478.8 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198478.8 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> SH3BGRL2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:15567 </td>
   <td style="text-align:left;"> OTTHUMG00000015081.3 </td>
   <td style="text-align:right;"> -4.077773 </td>
   <td style="text-align:right;"> 3.2297321 </td>
   <td style="text-align:right;"> -21.92270 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 166.1165 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000150672.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000150672.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> DLG2 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:2901 </td>
   <td style="text-align:left;"> OTTHUMG00000134309.9 </td>
   <td style="text-align:right;"> -3.472221 </td>
   <td style="text-align:right;"> -0.5579568 </td>
   <td style="text-align:right;"> -21.55086 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 161.5919 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000158458.21 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000158458.21 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> NRG2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:7998 </td>
   <td style="text-align:left;"> OTTHUMG00000129241.9 </td>
   <td style="text-align:right;"> -4.149858 </td>
   <td style="text-align:right;"> -2.0863801 </td>
   <td style="text-align:right;"> -21.53525 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 161.2810 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198643.7 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198643.7 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FAM3D </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:18665 </td>
   <td style="text-align:left;"> OTTHUMG00000159148.3 </td>
   <td style="text-align:right;"> -5.774163 </td>
   <td style="text-align:right;"> 0.8014869 </td>
   <td style="text-align:right;"> -21.47256 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 160.8004 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000152642.11 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000152642.11 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> GPD1L </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:28956 </td>
   <td style="text-align:left;"> OTTHUMG00000155846.3 </td>
   <td style="text-align:right;"> -2.638221 </td>
   <td style="text-align:right;"> 3.9623322 </td>
   <td style="text-align:right;"> -21.24362 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 158.0873 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000134240.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000134240.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> HMGCS2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:5008 </td>
   <td style="text-align:left;"> OTTHUMG00000012101.4 </td>
   <td style="text-align:right;"> -6.098245 </td>
   <td style="text-align:right;"> -3.4727073 </td>
   <td style="text-align:right;"> -21.04294 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 155.5405 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000094963.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000094963.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FMO2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3770 </td>
   <td style="text-align:left;"> OTTHUMG00000035504.8 </td>
   <td style="text-align:right;"> -5.166329 </td>
   <td style="text-align:right;"> 2.0436278 </td>
   <td style="text-align:right;"> -20.75100 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 152.2849 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000167676.4 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000167676.4 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> PLIN4 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:29393 </td>
   <td style="text-align:left;"> OTTHUMG00000167571.2 </td>
   <td style="text-align:right;"> -4.109265 </td>
   <td style="text-align:right;"> 1.0643126 </td>
   <td style="text-align:right;"> -20.51358 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 149.5047 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000184811.4 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000184811.4 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> TRARG1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:29592 </td>
   <td style="text-align:left;"> OTTHUMG00000132196.3 </td>
   <td style="text-align:right;"> -4.662262 </td>
   <td style="text-align:right;"> -4.5074038 </td>
   <td style="text-align:right;"> -20.04309 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 143.5722 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000187288.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000187288.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> CIDEC </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:24229 </td>
   <td style="text-align:left;"> OTTHUMG00000128522.4 </td>
   <td style="text-align:right;"> -4.908541 </td>
   <td style="text-align:right;"> -4.0234673 </td>
   <td style="text-align:right;"> -19.95167 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 142.6007 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000124205.18 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000124205.18 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> EDN3 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3178 </td>
   <td style="text-align:left;"> OTTHUMG00000032867.6 </td>
   <td style="text-align:right;"> -5.703333 </td>
   <td style="text-align:right;"> -3.7664117 </td>
   <td style="text-align:right;"> -19.40476 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 136.3662 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198099.9 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198099.9 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ADH4 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:252 </td>
   <td style="text-align:left;"> OTTHUMG00000161227.3 </td>
   <td style="text-align:right;"> -4.041885 </td>
   <td style="text-align:right;"> -4.6033099 </td>
   <td style="text-align:right;"> -19.31364 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 135.0729 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>Then by gender:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Pipeline top genes by gender_eb115f744f2ebd5394807221440bc8a2">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gender_limma_res<span class="sc">$</span>topGenes, <span class="dv">20</span>) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> source </th>
   <th style="text-align:left;"> type </th>
   <th style="text-align:right;"> score </th>
   <th style="text-align:right;"> phase </th>
   <th style="text-align:left;"> gene_id </th>
   <th style="text-align:left;"> gene_type </th>
   <th style="text-align:left;"> gene_name </th>
   <th style="text-align:left;"> level </th>
   <th style="text-align:left;"> hgnc_id </th>
   <th style="text-align:left;"> havana_gene </th>
   <th style="text-align:right;"> logFC </th>
   <th style="text-align:right;"> AveExpr </th>
   <th style="text-align:right;"> t </th>
   <th style="text-align:right;"> P.Value </th>
   <th style="text-align:right;"> adj.P.Val </th>
   <th style="text-align:right;"> B </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000229807.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000229807.12 </td>
   <td style="text-align:left;"> lncRNA </td>
   <td style="text-align:left;"> XIST </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:12810 </td>
   <td style="text-align:left;"> OTTHUMG00000021839.13 </td>
   <td style="text-align:right;"> -9.138110 </td>
   <td style="text-align:right;"> -2.2553619 </td>
   <td style="text-align:right;"> -57.01435 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 517.04453 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000165246.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000165246.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> NLGN4Y </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:15529 </td>
   <td style="text-align:left;"> OTTHUMG00000036618.4 </td>
   <td style="text-align:right;"> 7.858163 </td>
   <td style="text-align:right;"> 0.0569294 </td>
   <td style="text-align:right;"> 32.20743 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 275.78739 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000012817.16 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000012817.16 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> KDM5D </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:11115 </td>
   <td style="text-align:left;"> OTTHUMG00000036508.3 </td>
   <td style="text-align:right;"> 8.992662 </td>
   <td style="text-align:right;"> 1.4110791 </td>
   <td style="text-align:right;"> 30.29833 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 256.05307 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000067048.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000067048.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> DDX3Y </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:2699 </td>
   <td style="text-align:left;"> OTTHUMG00000036324.3 </td>
   <td style="text-align:right;"> 9.364232 </td>
   <td style="text-align:right;"> 2.1746779 </td>
   <td style="text-align:right;"> 29.55618 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 248.67737 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000131002.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000131002.12 </td>
   <td style="text-align:left;"> transcribed_unprocessed_pseudogene </td>
   <td style="text-align:left;"> TXLNGY </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:18473 </td>
   <td style="text-align:left;"> OTTHUMG00000036509.2 </td>
   <td style="text-align:right;"> 8.499072 </td>
   <td style="text-align:right;"> 0.5899586 </td>
   <td style="text-align:right;"> 29.38671 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 245.58451 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198692.10 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198692.10 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> EIF1AY </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3252 </td>
   <td style="text-align:left;"> OTTHUMG00000036544.3 </td>
   <td style="text-align:right;"> 8.722098 </td>
   <td style="text-align:right;"> 0.7437648 </td>
   <td style="text-align:right;"> 29.18213 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 243.38447 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000067646.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000067646.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ZFY </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:12870 </td>
   <td style="text-align:left;"> OTTHUMG00000036159.5 </td>
   <td style="text-align:right;"> 8.386303 </td>
   <td style="text-align:right;"> 0.3587296 </td>
   <td style="text-align:right;"> 29.10210 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 242.37404 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000183878.15 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000183878.15 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> UTY </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:12638 </td>
   <td style="text-align:left;"> OTTHUMG00000036319.3 </td>
   <td style="text-align:right;"> 8.365311 </td>
   <td style="text-align:right;"> 0.5707810 </td>
   <td style="text-align:right;"> 27.90240 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 229.25128 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000129824.16 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000129824.16 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> RPS4Y1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:10425 </td>
   <td style="text-align:left;"> OTTHUMG00000036152.5 </td>
   <td style="text-align:right;"> 9.654276 </td>
   <td style="text-align:right;"> 3.8514859 </td>
   <td style="text-align:right;"> 27.37539 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 226.08925 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000114374.13 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000114374.13 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> USP9Y </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:12633 </td>
   <td style="text-align:left;"> OTTHUMG00000036469.3 </td>
   <td style="text-align:right;"> 8.582255 </td>
   <td style="text-align:right;"> 1.1158829 </td>
   <td style="text-align:right;"> 26.49029 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 213.85041 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000099725.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000099725.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> PRKY </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:9444 </td>
   <td style="text-align:left;"> OTTHUMG00000035301.4 </td>
   <td style="text-align:right;"> 7.536312 </td>
   <td style="text-align:right;"> 0.5629283 </td>
   <td style="text-align:right;"> 26.17693 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 210.35811 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000260197.1 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000260197.1 </td>
   <td style="text-align:left;"> lncRNA </td>
   <td style="text-align:left;"> AC010889.1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:left;"> OTTHUMG00000175431.1 </td>
   <td style="text-align:right;"> 4.738207 </td>
   <td style="text-align:right;"> -3.1616839 </td>
   <td style="text-align:right;"> 25.86571 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 205.61120 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000215580.11 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000215580.11 </td>
   <td style="text-align:left;"> transcribed_unprocessed_pseudogene </td>
   <td style="text-align:left;"> BCORP1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:23953 </td>
   <td style="text-align:left;"> OTTHUMG00000036498.2 </td>
   <td style="text-align:right;"> 3.963650 </td>
   <td style="text-align:right;"> -3.4953728 </td>
   <td style="text-align:right;"> 23.15772 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 175.07142 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000176728.10 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000176728.10 </td>
   <td style="text-align:left;"> lncRNA </td>
   <td style="text-align:left;"> TTTY14 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:18495 </td>
   <td style="text-align:left;"> OTTHUMG00000036505.19 </td>
   <td style="text-align:right;"> 5.626969 </td>
   <td style="text-align:right;"> -2.0520471 </td>
   <td style="text-align:right;"> 23.01289 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 173.72703 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000231535.7 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000231535.7 </td>
   <td style="text-align:left;"> lncRNA </td>
   <td style="text-align:left;"> LINC00278 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:38712 </td>
   <td style="text-align:left;"> OTTHUMG00000036157.7 </td>
   <td style="text-align:right;"> 4.890089 </td>
   <td style="text-align:right;"> -3.0190605 </td>
   <td style="text-align:right;"> 22.59414 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 168.84636 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000286009.2 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000286009.2 </td>
   <td style="text-align:left;"> lncRNA </td>
   <td style="text-align:left;"> AC244213.1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:left;"> OTTHUMG00000193172.5 </td>
   <td style="text-align:right;"> 3.968396 </td>
   <td style="text-align:right;"> -3.6612343 </td>
   <td style="text-align:right;"> 21.47465 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 156.07335 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000154620.6 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000154620.6 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> TMSB4Y </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:11882 </td>
   <td style="text-align:left;"> OTTHUMG00000036366.2 </td>
   <td style="text-align:right;"> 4.664996 </td>
   <td style="text-align:right;"> -3.2390062 </td>
   <td style="text-align:right;"> 21.37609 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 155.06559 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000241859.7 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000241859.7 </td>
   <td style="text-align:left;"> transcribed_unprocessed_pseudogene </td>
   <td style="text-align:left;"> ANOS2P </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:6214 </td>
   <td style="text-align:left;"> OTTHUMG00000036365.4 </td>
   <td style="text-align:right;"> 4.243049 </td>
   <td style="text-align:right;"> -3.3932792 </td>
   <td style="text-align:right;"> 17.24732 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 109.18787 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000237659.1 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000237659.1 </td>
   <td style="text-align:left;"> processed_pseudogene </td>
   <td style="text-align:left;"> RNASEH2CP1 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:24117 </td>
   <td style="text-align:left;"> OTTHUMG00000036081.1 </td>
   <td style="text-align:right;"> 3.425815 </td>
   <td style="text-align:right;"> -4.1460728 </td>
   <td style="text-align:right;"> 15.25351 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 87.89118 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000184895.8 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000184895.8 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> SRY </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:11311 </td>
   <td style="text-align:left;"> OTTHUMG00000036084.2 </td>
   <td style="text-align:right;"> 3.079100 </td>
   <td style="text-align:right;"> -4.4075079 </td>
   <td style="text-align:right;"> 14.10067 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 76.07488 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>*<strong>Why don’t the genes that come out of the ‘manual’ and automatic analyses match here? I would have thought that they should both be identical but would have to examine the code of the automated block to be sure. (TO DO)</strong></p>
</section>
<section id="visualization" class="level2">
<h2 class="anchored" data-anchor-id="visualization">Visualization</h2>
<section id="pca-plot" class="level3">
<h3 class="anchored" data-anchor-id="pca-plot">PCA Plot</h3>
<p>The authors have prepared another function which uses the “voom” object that comes out of the limma pipeline <strong>To read about later, this is a “Large EList”.</strong> You then plug this into a principal components anysis (PCA) though the authors are light on an explanation of the variable that is being subjected to this analysis and I will have to read up on this further.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/PCA plot function_99699c324c00acff0cd5809031299ae0">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>plot_PCA <span class="ot">&lt;-</span> <span class="cf">function</span>(voomObj, condition_variable){</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>  group <span class="ot">=</span> <span class="fu">factor</span>(voomObj<span class="sc">$</span>targets[, condition_variable])</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  pca <span class="ot">=</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(voomObj<span class="sc">$</span>E))</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Take PC1 and PC2 for the plot</span></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="at">col=</span>group, <span class="at">pch=</span><span class="dv">19</span>)</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># include a legend for points</span></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend</span>(<span class="st">"bottomleft"</span>, <span class="at">inset=</span>.<span class="dv">01</span>, <span class="fu">levels</span>(group), <span class="at">pch=</span><span class="dv">19</span>, <span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(group)))</span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pca)</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>They then use the function <code>plot_PCA()</code> to get the first <strong>two</strong> principal components and plot them against each other with the conditional variable in this case being cancer vs non-cancer tissue.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/PCA plot of cancer vs normal_b684ebddfd9850625d5ee7ef2adf9e77">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>res_pca <span class="ot">&lt;-</span> <span class="fu">plot_PCA</span>(limma_res<span class="sc">$</span>voomObj, <span class="st">"definition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/PCA%20plot%20of%20cancer%20vs%20normal-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The usual problem of the explicability of the PCA model is exacerbated by the fact that I don’t have a good intuitive feel for the factors being analysed here (presumably DNA expression levels for all the genes in the human genome plus some other sequences included in the genome). Unlike the example here, there doesn’t appear to be great separation of the RNA expression profiles of the two groups. Out of curiosity I’m also going to have a look at the gender data. I expect better separation here due to factors like the SRY gene coming out of the analysis above:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/PCA plot of male vs female_5b5363c4ec5a8cc6758cd6901e834614">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>gender_res_pca <span class="ot">&lt;-</span> <span class="fu">plot_PCA</span>(gender_limma_res<span class="sc">$</span>voomObj, <span class="st">"definition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/PCA%20plot%20of%20male%20vs%20female-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Hmm, not fantastic, maybe the relatively low sample size of H&amp;N tumours in the TCGA database or heterogeneity between subsites is causing this?</p>
</section>
<section id="heatmap" class="level3">
<h3 class="anchored" data-anchor-id="heatmap">Heatmap</h3>
<p>Going back to the walk-through and focusing on tumour vs normal tissue, I will create a heatmap:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Selecting top 20 genes in sample_b3f8858d1102247b3fa363aab94ab742">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># first lets get the normalized expression matrix from our limma_res object</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(limma_res<span class="sc">$</span>voomObj<span class="sc">$</span>E))</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="co"># then lets get gene names that are easier to look at</span></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> limma_res<span class="sc">$</span>voomObj<span class="sc">$</span>genes[,<span class="st">"gene_name"</span>]</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a><span class="co"># and use these to rename the genes in our expression matrix</span></span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_mat) <span class="ot">&lt;-</span> gene_names</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a><span class="co"># we want to get the top20 DE genes</span></span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a><span class="co"># topGenes is already sorted by the adjusted p-value (in ascending order)</span></span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(limma_res<span class="sc">$</span>topGenes,<span class="dv">20</span>) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> source </th>
   <th style="text-align:left;"> type </th>
   <th style="text-align:right;"> score </th>
   <th style="text-align:right;"> phase </th>
   <th style="text-align:left;"> gene_id </th>
   <th style="text-align:left;"> gene_type </th>
   <th style="text-align:left;"> gene_name </th>
   <th style="text-align:left;"> level </th>
   <th style="text-align:left;"> hgnc_id </th>
   <th style="text-align:left;"> havana_gene </th>
   <th style="text-align:right;"> logFC </th>
   <th style="text-align:right;"> AveExpr </th>
   <th style="text-align:right;"> t </th>
   <th style="text-align:right;"> P.Value </th>
   <th style="text-align:right;"> adj.P.Val </th>
   <th style="text-align:right;"> B </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> ENSG00000181092.10 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000181092.10 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ADIPOQ </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:13633 </td>
   <td style="text-align:left;"> OTTHUMG00000156521.3 </td>
   <td style="text-align:right;"> -7.176794 </td>
   <td style="text-align:right;"> -4.7711913 </td>
   <td style="text-align:right;"> -27.13784 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 227.1603 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000096006.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000096006.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> CRISP3 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:16904 </td>
   <td style="text-align:left;"> OTTHUMG00000014823.3 </td>
   <td style="text-align:right;"> -8.591856 </td>
   <td style="text-align:right;"> -1.0019402 </td>
   <td style="text-align:right;"> -26.60984 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 221.6612 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000196616.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000196616.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ADH1B </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:250 </td>
   <td style="text-align:left;"> OTTHUMG00000161413.4 </td>
   <td style="text-align:right;"> -6.608842 </td>
   <td style="text-align:right;"> -2.3444701 </td>
   <td style="text-align:right;"> -24.26854 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 193.8269 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000034971.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000034971.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> MYOC </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:7610 </td>
   <td style="text-align:left;"> OTTHUMG00000034789.4 </td>
   <td style="text-align:right;"> -6.335138 </td>
   <td style="text-align:right;"> -4.4762472 </td>
   <td style="text-align:right;"> -23.98936 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 190.0965 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000102547.19 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000102547.19 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> CAB39L </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:20290 </td>
   <td style="text-align:left;"> OTTHUMG00000016914.6 </td>
   <td style="text-align:right;"> -2.238875 </td>
   <td style="text-align:right;"> 2.6534008 </td>
   <td style="text-align:right;"> -23.87185 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 189.2265 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000167588.13 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000167588.13 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> GPD1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:4455 </td>
   <td style="text-align:left;"> OTTHUMG00000169813.6 </td>
   <td style="text-align:right;"> -4.683988 </td>
   <td style="text-align:right;"> -0.4668095 </td>
   <td style="text-align:right;"> -23.86213 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 189.0425 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000166819.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000166819.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> PLIN1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:9076 </td>
   <td style="text-align:left;"> OTTHUMG00000149813.5 </td>
   <td style="text-align:right;"> -4.974732 </td>
   <td style="text-align:right;"> -2.4730633 </td>
   <td style="text-align:right;"> -22.73266 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 175.4408 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000168309.18 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000168309.18 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FAM107A </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:30827 </td>
   <td style="text-align:left;"> OTTHUMG00000159159.5 </td>
   <td style="text-align:right;"> -3.810493 </td>
   <td style="text-align:right;"> 1.1439306 </td>
   <td style="text-align:right;"> -22.47880 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 172.6963 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198478.8 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198478.8 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> SH3BGRL2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:15567 </td>
   <td style="text-align:left;"> OTTHUMG00000015081.3 </td>
   <td style="text-align:right;"> -4.077773 </td>
   <td style="text-align:right;"> 3.2297321 </td>
   <td style="text-align:right;"> -21.92270 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 166.1165 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000150672.17 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000150672.17 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> DLG2 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:2901 </td>
   <td style="text-align:left;"> OTTHUMG00000134309.9 </td>
   <td style="text-align:right;"> -3.472221 </td>
   <td style="text-align:right;"> -0.5579568 </td>
   <td style="text-align:right;"> -21.55086 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 161.5919 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000158458.21 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000158458.21 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> NRG2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:7998 </td>
   <td style="text-align:left;"> OTTHUMG00000129241.9 </td>
   <td style="text-align:right;"> -4.149858 </td>
   <td style="text-align:right;"> -2.0863801 </td>
   <td style="text-align:right;"> -21.53525 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 161.2810 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198643.7 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198643.7 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FAM3D </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:18665 </td>
   <td style="text-align:left;"> OTTHUMG00000159148.3 </td>
   <td style="text-align:right;"> -5.774163 </td>
   <td style="text-align:right;"> 0.8014869 </td>
   <td style="text-align:right;"> -21.47256 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 160.8004 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000152642.11 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000152642.11 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> GPD1L </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:28956 </td>
   <td style="text-align:left;"> OTTHUMG00000155846.3 </td>
   <td style="text-align:right;"> -2.638221 </td>
   <td style="text-align:right;"> 3.9623322 </td>
   <td style="text-align:right;"> -21.24362 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 158.0873 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000134240.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000134240.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> HMGCS2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:5008 </td>
   <td style="text-align:left;"> OTTHUMG00000012101.4 </td>
   <td style="text-align:right;"> -6.098245 </td>
   <td style="text-align:right;"> -3.4727073 </td>
   <td style="text-align:right;"> -21.04294 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 155.5405 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000094963.14 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000094963.14 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> FMO2 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3770 </td>
   <td style="text-align:left;"> OTTHUMG00000035504.8 </td>
   <td style="text-align:right;"> -5.166329 </td>
   <td style="text-align:right;"> 2.0436278 </td>
   <td style="text-align:right;"> -20.75100 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 152.2849 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000167676.4 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000167676.4 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> PLIN4 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:29393 </td>
   <td style="text-align:left;"> OTTHUMG00000167571.2 </td>
   <td style="text-align:right;"> -4.109265 </td>
   <td style="text-align:right;"> 1.0643126 </td>
   <td style="text-align:right;"> -20.51358 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 149.5047 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000184811.4 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000184811.4 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> TRARG1 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:29592 </td>
   <td style="text-align:left;"> OTTHUMG00000132196.3 </td>
   <td style="text-align:right;"> -4.662262 </td>
   <td style="text-align:right;"> -4.5074038 </td>
   <td style="text-align:right;"> -20.04309 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 143.5722 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000187288.12 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000187288.12 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> CIDEC </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:24229 </td>
   <td style="text-align:left;"> OTTHUMG00000128522.4 </td>
   <td style="text-align:right;"> -4.908541 </td>
   <td style="text-align:right;"> -4.0234673 </td>
   <td style="text-align:right;"> -19.95167 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 142.6007 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000124205.18 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000124205.18 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> EDN3 </td>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> HGNC:3178 </td>
   <td style="text-align:left;"> OTTHUMG00000032867.6 </td>
   <td style="text-align:right;"> -5.703333 </td>
   <td style="text-align:right;"> -3.7664117 </td>
   <td style="text-align:right;"> -19.40476 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 136.3662 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> ENSG00000198099.9 </td>
   <td style="text-align:left;"> HAVANA </td>
   <td style="text-align:left;"> gene </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:left;"> ENSG00000198099.9 </td>
   <td style="text-align:left;"> protein_coding </td>
   <td style="text-align:left;"> ADH4 </td>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> HGNC:252 </td>
   <td style="text-align:left;"> OTTHUMG00000161227.3 </td>
   <td style="text-align:right;"> -4.041885 </td>
   <td style="text-align:right;"> -4.6033099 </td>
   <td style="text-align:right;"> -19.31364 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 135.0729 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Creating HeatMap_2fdc7ff7ca6347fb487c41557f935392">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># therefore we can just go ahead and select the first 20 values</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>top_de_genes <span class="ot">&lt;-</span> limma_res<span class="sc">$</span>topGenes<span class="sc">$</span>gene_name[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="co"># we'll get the sample type data so that we can show the groups in the heatmap</span></span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>sample_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(limma_res<span class="sc">$</span>voomObj<span class="sc">$</span>targets<span class="sc">$</span>sample_type)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="co"># define the color palette for the plot</span></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>hmcol <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"RdBu"</span>)))(<span class="dv">256</span>)</span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a><span class="co"># perform complete linkage clustering</span></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>clust_func <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">hclust</span>(x, <span class="at">method=</span><span class="st">"complete"</span>)</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a><span class="co"># use the inverse of correlation as distance.</span></span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>dist_func <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">as.dist</span>((<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">t</span>(x)))<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a><span class="co"># A good looking heatmap involves a lot of parameters and tinkering</span></span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a>gene_heatmap <span class="ot">&lt;-</span> <span class="fu">heatmap.2</span>(</span>
<span id="cb93-18"><a href="#cb93-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(expr_mat[,top_de_genes]),</span>
<span id="cb93-19"><a href="#cb93-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale=</span><span class="st">"row"</span>,          <span class="co"># scale the values for each gene (row)</span></span>
<span id="cb93-20"><a href="#cb93-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">density.info=</span><span class="st">"none"</span>,  <span class="co"># turns off density plot inside color legend</span></span>
<span id="cb93-21"><a href="#cb93-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">trace=</span><span class="st">"none"</span>,         <span class="co"># turns off trace lines inside the heat map</span></span>
<span id="cb93-22"><a href="#cb93-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">col=</span>hmcol,            <span class="co"># define the color map</span></span>
<span id="cb93-23"><a href="#cb93-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">labCol=</span><span class="cn">FALSE</span>,         <span class="co"># Not showing column labels</span></span>
<span id="cb93-24"><a href="#cb93-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">ColSideColors=</span><span class="fu">as.character</span>(<span class="fu">as.numeric</span>(sample_type)), <span class="co"># Show colors for each response class</span></span>
<span id="cb93-25"><a href="#cb93-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendrogram=</span><span class="st">"both"</span>,    <span class="co"># Show dendrograms for both axis</span></span>
<span id="cb93-26"><a href="#cb93-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">hclust =</span> clust_func,  <span class="co"># Define hierarchical clustering method</span></span>
<span id="cb93-27"><a href="#cb93-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">distfun =</span> dist_func,  <span class="co"># Using correlation coefficient for distance function</span></span>
<span id="cb93-28"><a href="#cb93-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">cexRow=</span><span class="fl">1.0</span>,           <span class="co"># Resize row labels</span></span>
<span id="cb93-29"><a href="#cb93-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">keysize =</span> <span class="fl">1.25</span>,       <span class="co"># Size of the legend</span></span>
<span id="cb93-30"><a href="#cb93-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">margins=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>)        <span class="co"># Define margin spaces</span></span>
<span id="cb93-31"><a href="#cb93-31" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Creating%20HeatMap-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="interpreting-heatmap" class="level3">
<h3 class="anchored" data-anchor-id="interpreting-heatmap">Interpreting HeatMap</h3>
<ul>
<li>This Heat Map is based on The genes on the vertical column with the samples in the horizontal column. I cannot yet interpret the treee <strong>come back to this</strong></li>
</ul>
</section>
</section>
<section id="further-heatmap-exercise" class="level2">
<h2 class="anchored" data-anchor-id="further-heatmap-exercise">Further heatmap exercise</h2>
<p>The exercise set is to produce a heatmap for gender data. It would be relatively simple to repeat the above code but this would not be a “DRY” method. In order to parametrise the above operations I need to figure out how to select the right limma object and comparators. The limma object itself sets some of these factors. I need to be able to pass these elements into a function as arguments. Hence, I need a function.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Functionalising pipeline further_34c44521f53e00c11757347f671d22cf">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># YOU HAVE NOT PUT THE COMPARATOR IN THE FUNCTION</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="co"># THIS FUNCITON NEEDS THE LIMMA PIPELINE ADDED AND THEN TO PASS IN THE COMPARATOR</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>my.heatmap.fun <span class="ot">&lt;-</span> <span class="cf">function</span>(data_source, comparator, number_of_genes) {</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>  top_genes_in_result <span class="ot">&lt;-</span>  data_source <span class="sc">%&gt;%</span> </span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>    magrittr<span class="sc">::</span><span class="fu">extract2</span>(<span class="st">"topGenes"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> number_of_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>()</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>  type_of_sample <span class="ot">&lt;-</span> data_source <span class="sc">%&gt;%</span></span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>    magrittr<span class="sc">::</span><span class="fu">use_series</span>(<span class="st">"voomObj"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>    magrittr<span class="sc">::</span><span class="fu">use_series</span>(<span class="st">"targets"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>    magrittr<span class="sc">::</span><span class="fu">use_series</span>(<span class="st">"sample_type"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>()  </span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>  hmcol <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"RdBu"</span>)))(<span class="dv">256</span>)</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a>  clust_func <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">hclust</span>(x, <span class="at">method=</span><span class="st">"complete"</span>)</span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a>  dist_func <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">as.dist</span>((<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">t</span>(x)))<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb94-23"><a href="#cb94-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-24"><a href="#cb94-24" aria-hidden="true" tabindex="-1"></a>  gene_heatmap <span class="ot">&lt;-</span> <span class="fu">heatmap.2</span>(</span>
<span id="cb94-25"><a href="#cb94-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(expr_mat[,top_genes_in_result]),</span>
<span id="cb94-26"><a href="#cb94-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale=</span><span class="st">"row"</span>,          <span class="co"># scale the values for each gene (row)</span></span>
<span id="cb94-27"><a href="#cb94-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">density.info=</span><span class="st">"none"</span>,  <span class="co"># turns off density plot inside color legend</span></span>
<span id="cb94-28"><a href="#cb94-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">trace=</span><span class="st">"none"</span>,         <span class="co"># turns off trace lines inside the heat map</span></span>
<span id="cb94-29"><a href="#cb94-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">col=</span>hmcol,            <span class="co"># define the color map</span></span>
<span id="cb94-30"><a href="#cb94-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">labCol=</span><span class="cn">FALSE</span>,         <span class="co"># Not showing column labels</span></span>
<span id="cb94-31"><a href="#cb94-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">ColSideColors=</span><span class="fu">as.character</span>(<span class="fu">as.numeric</span>(type_of_sample)), <span class="co"># Show colors for each response class</span></span>
<span id="cb94-32"><a href="#cb94-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendrogram=</span><span class="st">"both"</span>,    <span class="co"># Show dendrograms for both axis</span></span>
<span id="cb94-33"><a href="#cb94-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">hclust =</span> clust_func,  <span class="co"># Define hierarchical clustering method</span></span>
<span id="cb94-34"><a href="#cb94-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">distfun =</span> dist_func,  <span class="co"># Using correlation coefficient for distance function</span></span>
<span id="cb94-35"><a href="#cb94-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">cexRow=</span><span class="fl">1.0</span>,           <span class="co"># Resize row labels</span></span>
<span id="cb94-36"><a href="#cb94-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">keysize =</span> <span class="fl">1.25</span>,       <span class="co"># Size of the legend</span></span>
<span id="cb94-37"><a href="#cb94-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">margins=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>)        <span class="co"># Define margin spaces</span></span>
<span id="cb94-38"><a href="#cb94-38" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb94-39"><a href="#cb94-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb94-40"><a href="#cb94-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-41"><a href="#cb94-41" aria-hidden="true" tabindex="-1"></a><span class="fu">my.heatmap.fun</span>(limma_res, sample_type, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Functionalising%20pipeline%20further-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">my.heatmap.fun</span>(gender_limma_res, sample_type, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Functionalising%20pipeline%20further-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(colData(tcga_data))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>TO DO: Come back and try different factors such as T stage or N stage</p>
<p>Questions: 1. What difference in the main tumour predicts nodal spread? 1. What role does stage play? 1. Do these correlate with survival? 1. What other factors (such as age) are important? 1. Can I generate a table 1 for these data? 1. Can I look at smoking in this dataset in ways that haven’t been published already?</p>
<p>Moving on with the exercise, let’s get through the survival exercises so that we can apply these to the questions we want to answer.</p>
</section>
<section id="survival" class="level2">
<h2 class="anchored" data-anchor-id="survival">Survival</h2>
<section id="survival-by-gender" class="level3">
<h3 class="anchored" data-anchor-id="survival-by-gender">Survival by gender</h3>
<p>In the example, they look at gender and ask whether this predicts survival. We have already created a relevant object above for this where gene expression is concerned. Now let’s see what we can do for survival.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Extracting clinical data from tcga_data_590c3fb84c4a0d107974bcc551193ccd">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>clinical <span class="ot">&lt;-</span> tcga_data<span class="sc">@</span>colData</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="co"># We only want the "Primary solid Tumor" cases here</span></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>clin_df <span class="ot">&lt;-</span> clinical[clinical<span class="sc">$</span>definition <span class="sc">==</span> <span class="st">"Primary solid Tumor"</span>, </span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">c</span>(<span class="st">"patient"</span>, </span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"vital_status"</span>,</span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"days_to_death"</span>,</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"days_to_last_follow_up"</span>,</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"gender"</span>,</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"ajcc_pathologic_stage"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The exercise authors have selected out only the columns that are relevant to survival analysis here. I did try to replicate the above using dplyr functions but simply cannot get these to work with the S4 DFrame that bioconductor uses here.</p>
<p>Interestingly, the exercise recommends creating a new column for the survival status using a boolean rather than a 1 or zero as I have previously used in the survival package:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Defining Event columm_c4512dc714138e121d222c851e8d0e5c">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>clin_df<span class="sc">$</span>deceased <span class="ot">&lt;-</span> clin_df<span class="sc">$</span>vital_status <span class="sc">==</span> <span class="st">"Dead"</span> <span class="co"># This creates TRUE for dead and FALSE for alive</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In the current frame, time to last follow-up and time to death are in separate columns. In survival analysis, we need to know the time to the last contact with the patient whether this be death or follow-up and the reelvant time needs to be in the same column:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Defining time to event/ last f/u_b54b2ab3c65677185ba5aff907e1cbba">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>clin_df<span class="sc">$</span>overall_survival <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(clin_df<span class="sc">$</span>deceased, <span class="co"># Shorthand for clin_df$deceased == TRUE</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>                                   clin_df<span class="sc">$</span>overall_survival <span class="ot">&lt;-</span> clin_df<span class="sc">$</span>days_to_death,</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>                                   clin_df<span class="sc">$</span>overall_survival <span class="ot">&lt;-</span> clin_df<span class="sc">$</span>days_to_last_follow_up)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can then create our fit using the survfit and Surv functions:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Kaplan Meier of gender versus survival_af8453adb130eaf79762649f61414e2f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(overall_survival, deceased) <span class="sc">~</span> gender, <span class="at">data=</span>clin_df)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Call: survfit(formula = Surv(overall_survival, deceased) ~ gender, 
    data = clin_df)

   2 observations deleted due to missingness 
                n events median 0.95LCL 0.95UCL
gender=female 135     69   1037     804    2166
gender=male   383    151   1838    1504    2570</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit, <span class="at">data =</span> clin_df, <span class="at">pval =</span> <span class="cn">TRUE</span>, <span class="at">risk.table =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Kaplan%20Meier%20of%20gender%20versus%20survival-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Although the p value is significant we can see that there is a lot of crossover of curves here suggesting that some of the Kaplan Meier assumptions have been violated and that this analysis may not be valid. Note that the syntax for the risk table seems to have changed since the example was written and I have updated this to work</p>
</section>
<section id="survival-by-stage" class="level3">
<h3 class="anchored" data-anchor-id="survival-by-stage">Survival by stage</h3>
<p>The example contains a very nice example of how to simplify staging that I can cross-apply to other data such as the GDO dataset:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Stage versus survival prep_ecbfc103b64d6b1edc620c3af46b085f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any of the letters "a", "b" or "c", but only if they are at the end</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="co"># of the name, eg "stage iiia" would become simply "stage iii"</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>clin_df<span class="sc">$</span>ajcc_pathologic_stage <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"[ABC]$"</span>, <span class="st">""</span>, clin_df<span class="sc">$</span>ajcc_pathologic_stage)</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a><span class="co"># let's see how many patients we have for each stage</span></span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">table</span>(clin_df<span class="sc">$</span>ajcc_pathologic_stage))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
  Stage I  Stage II Stage III  Stage IV 
       27        71        81       266 </code></pre>
</div>
</div>
<p>In the example they find too few stage IV patients and lump them with the stage IIIs. In this case it is not a problem but I <strong>should</strong> go back and remove the AJCC8 patients as these will be confounded by p16. I can now look at survival by stage. The example overwrites its fit which I dislike as it creates ambiguity as to what the object represents depending on run order when analysing in the editor so I create a new object for this</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/KM stage versus survival_f3ab464e52cad17e13d2fd2f00e7d312">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(overall_survival, deceased) <span class="sc">~</span> ajcc_pathologic_stage, <span class="at">data=</span>clin_df)</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(fit2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Call: survfit(formula = Surv(overall_survival, deceased) ~ ajcc_pathologic_stage, 
    data = clin_df)

   77 observations deleted due to missingness 
                                  n events median 0.95LCL 0.95UCL
ajcc_pathologic_stage=Stage I    27      4     NA      NA      NA
ajcc_pathologic_stage=Stage II   70     26   2570    1459      NA
ajcc_pathologic_stage=Stage III  81     29   1718    1430      NA
ajcc_pathologic_stage=Stage IV  265    130   1090     882    1732</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit2, <span class="at">data =</span> clin_df, <span class="at">pval =</span> <span class="cn">TRUE</span>, <span class="at">risk.table =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/KM%20stage%20versus%20survival-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>TO DO: filter out by oropharyngeal and AJCC7</p>
<p>As the example rightly states, the power of the TCGA dataset is its gene regulation data. We have disregarded this in the above survival analysis having figured out how to identify differentially expressed genes previously (heatmap, PCA etc.). We are now going to marry the two analyses to look at survival by gene.</p>
</section>
</section>
<section id="the-relationship-between-gene-expression-and-survival" class="level2">
<h2 class="anchored" data-anchor-id="the-relationship-between-gene-expression-and-survival">The relationship between gene expression and survival</h2>
<p>Earlier we looked at the top differentially expressed genes between the relevant states. In this case it was mainly tumour vs normal tissue. This returns the most differentially expressed genes (I believe that the default is 100 though we later filtered to just show the top 20 and thereby make graphs more interpretable).</p>
<p>Now we will look at the top gene and then see whether this relates to survival.</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Select top gene_38f2ca7b301ecdcef29fc14be0d232c7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>genes_df <span class="ot">&lt;-</span> limma_res<span class="sc">$</span>topGenes</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="co"># genes_df$gene_name[1] </span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>genes_df[<span class="dv">1</span>, <span class="st">"gene_name"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ADIPOQ"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the ensembl gene id of the first row</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>gene_id <span class="ot">&lt;-</span> genes_df[<span class="dv">1</span>, <span class="st">"gene_id"</span>]</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a><span class="co"># also get the common gene name of the first row</span></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>gene_name <span class="ot">&lt;-</span> genes_df[<span class="dv">1</span>, <span class="st">"gene_name"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="getting-expression-results-from-the-limma-result" class="level3">
<h3 class="anchored" data-anchor-id="getting-expression-results-from-the-limma-result">Getting expression results from the limma result</h3>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/unnamed-chunk-35_07257fbc4fa127281756eb7e7d97becb">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the actual expression matrix from the limma result object.</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="co"># we transpose it because we want genes in the columns and patients in the rows</span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>expr_df <span class="ot">&lt;-</span> <span class="fu">t</span>(limma_res<span class="sc">$</span>voomObj<span class="sc">$</span>E)</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a><span class="co"># let's show the first five genes for the first five patients</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>expr_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                             ENSG00000000003.15 ENSG00000000005.6
TCGA-D6-A4ZB-01A-11R-A24Z-07           4.684327         -6.580702
TCGA-CV-7435-01A-11R-2132-07           5.272325         -3.804194
TCGA-CR-7370-01A-11R-2132-07           5.724934         -7.406441
TCGA-HL-7533-01A-11R-2232-07           4.283945         -6.720978
TCGA-CR-7398-01A-11R-2016-07           5.712442         -5.555320
                             ENSG00000000419.13 ENSG00000000457.14
TCGA-D6-A4ZB-01A-11R-A24Z-07           4.728205           2.971967
TCGA-CV-7435-01A-11R-2132-07           4.718061           3.246260
TCGA-CR-7370-01A-11R-2132-07           4.047344           3.740128
TCGA-HL-7533-01A-11R-2232-07           5.300349           3.511443
TCGA-CR-7398-01A-11R-2016-07           4.827304           3.178260
                             ENSG00000000460.17
TCGA-D6-A4ZB-01A-11R-A24Z-07           2.358877
TCGA-CV-7435-01A-11R-2132-07           2.646101
TCGA-CR-7370-01A-11R-2132-07           4.041159
TCGA-HL-7533-01A-11R-2232-07           4.310379
TCGA-CR-7398-01A-11R-2016-07           3.762846</code></pre>
</div>
</div>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Plot differential expression_ff1e5825ff1a5841b94e1c865c4a955c">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize the gene expression distribution on the diseased samples (in black)</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="co"># versus the healthy samples (in red)</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>expr_diseased <span class="ot">&lt;-</span> expr_df[<span class="fu">rownames</span>(clin_df), gene_id]</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>expr_healthy <span class="ot">&lt;-</span> expr_df[<span class="fu">setdiff</span>(<span class="fu">rownames</span>(expr_df), <span class="fu">rownames</span>(clin_df)), gene_id]</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(expr_diseased, expr_healthy,</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"Diseased"</span>, <span class="st">"Healthy"</span>), </span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">main =</span> <span class="st">"Distribution of gene expression"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Plot%20differential%20expression-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># I want to see this with stats</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>expr_diseased2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(expr_diseased)</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>expr_diseased2<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="st">"Tumour"</span></span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_diseased2)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"expr"</span></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>expr_healthy2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(expr_healthy)</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>expr_healthy2<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="st">"Normal Tissue"</span></span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expr_healthy2)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"expr"</span></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>expr_for_plot <span class="ot">&lt;-</span> <span class="fu">rbind</span>(expr_diseased2, expr_healthy2)</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Putting tumour level first to make plot look better (avoid clash with stats)</span></span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>expr_for_plot<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">factor</span>(expr_for_plot<span class="sc">$</span>label,</span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>                              <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Tumour"</span>, <span class="st">"Normal Tissue"</span>))</span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> expr_for_plot, <span class="fu">aes</span>(<span class="at">x =</span> label, <span class="at">y =</span> expr)) <span class="sc">+</span></span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">colour =</span> <span class="st">"midnight blue"</span>) <span class="sc">+</span></span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>() <span class="sc">+</span></span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Comparison of"</span>, gene_name, <span class="st">"expression between groups"</span>),</span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Expression level"</span>, <span class="at">x =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Plot%20differential%20expression-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Wow, that was a lot of wrangling simplhy to get this into ggplot. </span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="co"># I need to figure out how to do this more smoothly without spending so much time</span></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a><span class="co"># getting data from wide to long!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The above tells us that on average diseased samples have a lower expression of the ADIPOQ gene but does not tell us whether the gene’s expression level has an effect on prognosis. Lets turn this around, in that case, to be able to provide that answer.</p>
</section>
<section id="influence-of-expression-on-prognosis" class="level3">
<h3 class="anchored" data-anchor-id="influence-of-expression-on-prognosis">Influence of expression on prognosis</h3>
<p>This is done simply with the median value of expression of the gene of interest</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Survival based on more or less than median expression_d7b2b57207c33b810437538a466ccd98">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the expression values for the selected gene</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>clin_df<span class="sc">$</span>gene_value <span class="ot">&lt;-</span> expr_df[<span class="fu">rownames</span>(clin_df), gene_id] <span class="co"># Note that this only selects diseased!</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a><span class="co"># find the median value of the gene and print it</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>median_value <span class="ot">&lt;-</span> <span class="fu">median</span>(clin_df<span class="sc">$</span>gene_value)</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>median_value</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -6.33046</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># divide patients in two groups, up and down regulated.</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="co"># if the patient expression is greater or equal to them median we put it</span></span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="co"># among the "up-regulated", otherwise among the "down-regulated"</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>clin_df<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(clin_df<span class="sc">$</span>gene_value <span class="sc">&gt;=</span> median_value, <span class="st">"UP"</span>, <span class="st">"DOWN"</span>)</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a><span class="co"># we can fit a survival model, like we did in the previous section</span></span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(overall_survival, deceased) <span class="sc">~</span> gene, <span class="at">data=</span>clin_df)</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a><span class="co"># we can extract the survival p-value and print it</span></span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>pval <span class="ot">&lt;-</span> <span class="fu">surv_pvalue</span>(fit, <span class="at">data=</span>clin_df)<span class="sc">$</span>pval</span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a>pval</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.9464799</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(fit, <span class="at">data=</span>clin_df, <span class="at">pval=</span>T, <span class="at">risk.table=</span>T, <span class="at">title=</span><span class="fu">paste</span>(gene_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Survival%20based%20on%20more%20or%20less%20than%20median%20expression-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There is stunningly little difference here. Note that the above code removed the normal tissue samples from the list (as it should) so we’re comparing surviaval between cancer patients only i.e.&nbsp;this tells us about the differential effect of the expression level on a gene on survival between cancer patients. This does not tell us how important the gene is compared to a normal patient i.e.&nbsp;if a gene was really important in defining a cancer cell it wouldn’t necessarily matter how expressed it was if ALL cancer patients had differential expression to normal tissue!</p>
</section>
</section>
<section id="functionalising-the-effect-of-expression-on-survival-for-multiple-genes" class="level2">
<h2 class="anchored" data-anchor-id="functionalising-the-effect-of-expression-on-survival-for-multiple-genes">Functionalising the effect of expression on survival for multiple genes</h2>
<p>Looking at the top ten genes that came out of my analysis <em>before</em> Benjamini-Hochberg adjustment for multiple tests:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Top 10 genes copied from exercies_d94fd0f21cdbf77e37bba67be5bf23c6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>) {</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get the ensembl gene id of the i-th row</span></span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>    gene_id <span class="ot">&lt;-</span> genes_df[i, <span class="st">"gene_id"</span>]</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># also get the common gene name of the i-th row</span></span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>    gene_name <span class="ot">&lt;-</span> genes_df[i, <span class="st">"gene_name"</span>]</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get the expression values for the selected gene</span></span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a>    clin_df<span class="sc">$</span>gene_value <span class="ot">=</span> expr_df[<span class="fu">rownames</span>(clin_df), gene_id]</span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the median value of the gene</span></span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a>    median_value <span class="ot">=</span> <span class="fu">median</span>(clin_df<span class="sc">$</span>gene_value)</span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># divide patients in two groups, up and down regulated.</span></span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if the patient expression is greater or equal to them median we put it</span></span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># among the "up-regulated", otherwise among the "down-regulated"</span></span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>    clin_df<span class="sc">$</span>gene <span class="ot">=</span> <span class="fu">ifelse</span>(clin_df<span class="sc">$</span>gene_value <span class="sc">&gt;=</span> median_value, <span class="st">"UP"</span>, <span class="st">"DOWN"</span>)</span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-19"><a href="#cb121-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># we can now run survival like we did in the previous section</span></span>
<span id="cb121-20"><a href="#cb121-20" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">=</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(overall_survival, deceased) <span class="sc">~</span> gene, <span class="at">data=</span>clin_df)</span>
<span id="cb121-21"><a href="#cb121-21" aria-hidden="true" tabindex="-1"></a>    survp <span class="ot">=</span> <span class="fu">ggsurvplot</span>(fit, <span class="at">data=</span>clin_df, <span class="at">pval=</span>T, <span class="at">risk.table=</span>T, <span class="at">title=</span><span class="fu">paste</span>(gene_name))</span>
<span id="cb121-22"><a href="#cb121-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(survp) </span>
<span id="cb121-23"><a href="#cb121-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># we can save the survival plot to file, try it by uncommenting it</span></span>
<span id="cb121-24"><a href="#cb121-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ggsave(file=paste0(gene_name, ".pdf"), print(survp))</span></span>
<span id="cb121-25"><a href="#cb121-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-26"><a href="#cb121-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># let's also save the p-value for later.. you'll see why :)</span></span>
<span id="cb121-27"><a href="#cb121-27" aria-hidden="true" tabindex="-1"></a>    genes_df[i, <span class="st">"surv_pv"</span>] <span class="ot">=</span> <span class="fu">surv_pvalue</span>(fit, <span class="at">data=</span>clin_df)<span class="sc">$</span>pval</span>
<span id="cb121-28"><a href="#cb121-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%2010%20genes%20copied%20from%20exercies-10.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Applying the Benjamini-Hochberg correction to correct the p values</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/unnamed-chunk-39_986d8ea81a334b8caec4dcd7c8c90db0">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">p.adjust</span>(genes_df<span class="sc">$</span>surv_pv, <span class="at">method =</span> <span class="st">"BH"</span>) <span class="co"># default length is length(genes_df$surv_pv)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] 0.9897940 0.9897940 0.4909744 0.7051940 0.4200192 0.9897940 0.5824860
  [8] 0.4200192 0.4200192 0.4200192        NA        NA        NA        NA
 [15]        NA        NA        NA        NA        NA        NA        NA
 [22]        NA        NA        NA        NA        NA        NA        NA
 [29]        NA        NA        NA        NA        NA        NA        NA
 [36]        NA        NA        NA        NA        NA        NA        NA
 [43]        NA        NA        NA        NA        NA        NA        NA
 [50]        NA        NA        NA        NA        NA        NA        NA
 [57]        NA        NA        NA        NA        NA        NA        NA
 [64]        NA        NA        NA        NA        NA        NA        NA
 [71]        NA        NA        NA        NA        NA        NA        NA
 [78]        NA        NA        NA        NA        NA        NA        NA
 [85]        NA        NA        NA        NA        NA        NA        NA
 [92]        NA        NA        NA        NA        NA        NA        NA
 [99]        NA        NA</code></pre>
</div>
</div>
</section>
<section id="understanding-results" class="level2">
<h2 class="anchored" data-anchor-id="understanding-results">Understanding results</h2>
<p>The above results tell us that the top 10 genes that came out of the initial analysis of normal versus cancer tissue are non-predictive of survival. I think that this is correct. I have to check that this is a true comparison of normal vs diseased tissue. Though this is important, I think that differential expressions between cancers themselves may be of interest. The problem is that any analysis in this dataset that segregated genes by survival would not then be testable unless we kept behind a test set (i.e.&nbsp;analyse a training set and then trial on a test set). This kind of approach is probably best kept for after the classification section of hte exercise or even trialling different test methods prior to ‘wasting’ the data on this sort of analysis. Either way, I am curious as to whether any ot the ‘Top Genes’ which initially contained 100 gene names carries any survival prediction so will run the above loop for the top 100 genes:</p>
<div class="cell" data-hash="TCGA_HN_Analysis_cache/html/Top 100 genes copied from exercies_f096f5d2f76ee8b9dd3d08fee01afe3a">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>) {</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get the ensembl gene id of the i-th row</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>    gene_id <span class="ot">&lt;-</span> genes_df[i, <span class="st">"gene_id"</span>]</span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># also get the common gene name of the i-th row</span></span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>    gene_name <span class="ot">&lt;-</span> genes_df[i, <span class="st">"gene_name"</span>]</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get the expression values for the selected gene</span></span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>    clin_df<span class="sc">$</span>gene_value <span class="ot">=</span> expr_df[<span class="fu">rownames</span>(clin_df), gene_id]</span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the median value of the gene</span></span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>    median_value <span class="ot">=</span> <span class="fu">median</span>(clin_df<span class="sc">$</span>gene_value)</span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># divide patients in two groups, up and down regulated.</span></span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if the patient expression is greater or equal to them median we put it</span></span>
<span id="cb124-16"><a href="#cb124-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># among the "up-regulated", otherwise among the "down-regulated"</span></span>
<span id="cb124-17"><a href="#cb124-17" aria-hidden="true" tabindex="-1"></a>    clin_df<span class="sc">$</span>gene <span class="ot">=</span> <span class="fu">ifelse</span>(clin_df<span class="sc">$</span>gene_value <span class="sc">&gt;=</span> median_value, <span class="st">"UP"</span>, <span class="st">"DOWN"</span>)</span>
<span id="cb124-18"><a href="#cb124-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-19"><a href="#cb124-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># we can now run survival like we did in the previous section</span></span>
<span id="cb124-20"><a href="#cb124-20" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">=</span> <span class="fu">survfit</span>(<span class="fu">Surv</span>(overall_survival, deceased) <span class="sc">~</span> gene, <span class="at">data=</span>clin_df)</span>
<span id="cb124-21"><a href="#cb124-21" aria-hidden="true" tabindex="-1"></a>    survp <span class="ot">=</span> <span class="fu">ggsurvplot</span>(fit, <span class="at">data=</span>clin_df, <span class="at">pval=</span>T, <span class="at">risk.table=</span>T, <span class="at">title=</span><span class="fu">paste</span>(gene_name))</span>
<span id="cb124-22"><a href="#cb124-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(survp) </span>
<span id="cb124-23"><a href="#cb124-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># we can save the survival plot to file, try it by uncommenting it</span></span>
<span id="cb124-24"><a href="#cb124-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ggsave(file=paste0(gene_name, ".pdf"), print(survp))</span></span>
<span id="cb124-25"><a href="#cb124-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-26"><a href="#cb124-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># let's also save the p-value for later.. you'll see why :)</span></span>
<span id="cb124-27"><a href="#cb124-27" aria-hidden="true" tabindex="-1"></a>    genes_df[i, <span class="st">"surv_pv"</span>] <span class="ot">=</span> <span class="fu">surv_pvalue</span>(fit, <span class="at">data=</span>clin_df)<span class="sc">$</span>pval</span>
<span id="cb124-28"><a href="#cb124-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-12.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-13.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-14.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-15.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-16.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-17.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-18.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-19.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-20.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-21.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-22.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-23.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-24.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-25.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-26.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-27.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-28.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-29.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-30.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-31.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-32.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-33.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-34.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-35.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-36.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-37.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-38.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-39.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-40.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-41.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-42.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-43.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-44.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-45.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-46.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-47.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-48.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-49.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-50.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-51.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-52.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-53.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-54.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-55.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-56.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-57.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-58.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-59.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-60.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-61.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-62.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-63.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-64.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-65.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-66.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-67.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-68.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-69.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-70.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-71.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-72.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-73.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-74.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-75.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-76.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-77.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-78.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-79.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-80.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-81.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-82.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-83.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-84.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-85.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-86.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-87.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-88.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-89.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-90.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-91.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-92.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-93.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-94.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-95.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-96.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-97.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-98.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-99.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="TCGA_HN_Analysis_files/figure-html/Top%20100%20genes%20copied%20from%20exercies-100.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">p.adjust</span>(genes_df<span class="sc">$</span>surv_pv, <span class="at">method =</span> <span class="st">"BH"</span>) <span class="co"># default length is length(genes_df$surv_pv)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] 0.98657910 0.98979395 0.62975724 0.88505099 0.50911413 0.98657910
  [7] 0.74095512 0.49305885 0.49305885 0.49305885 0.56798562 0.19176966
 [13] 0.98657910 0.98657910 0.49305885 0.88505099 0.88505099 0.95765620
 [19] 0.98657910 0.49305885 0.49305885 0.49305885 0.61113494 0.88505099
 [25] 0.98657910 0.88505099 0.05393488 0.98657910 0.65351167 0.98657910
 [31] 0.98657910 0.34267518 0.49305885 0.39990757 0.95765620 0.94366563
 [37] 0.49305885 0.16363758 0.49305885 0.98657910 0.88505099 0.77944160
 [43] 0.98657910 0.74095512 0.56914268 0.80453321 0.98657910 0.43086681
 [49] 0.70162170 0.98657910 0.98657910 0.65351167 0.98657910 0.06261349
 [55] 0.98657910 0.65111975 0.49305885 0.98657910 0.98657910 0.98657910
 [61] 0.56914268 0.24141024 0.49305885 0.98657910 0.98657910 0.16175711
 [67] 0.88844519 0.65351167 0.81415304 0.49305885 0.41810961 0.49305885
 [73] 0.49305885 0.49305885 0.96215849 0.95765620 0.49305885 0.05393488
 [79] 0.98657910 0.49305885 0.98657910 0.98657910 0.49305885 0.65111975
 [85] 0.24141024 0.98657910 0.98657910 0.74095512 0.98657910 0.88505099
 [91] 0.88844519 0.88505099 0.98657910 0.95765620 0.49305885 0.98657910
 [97] 0.98657910 0.71557241 0.98657910 0.62975724</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>